### ISSUE TITLE
when a downed alien ship falls, game crash

### ISSUE THREAD

**Issue by kayort** [OpenApoc/OpenApoc#1486]
[save.zip](https://github.com/user-attachments/files/16244928/save.zip)
[log.txt](https://github.com/user-attachments/files/16244942/log.txt)

Windows 10 x64 
OpenApoc-x64-20240331

W void __cdecl OpenApoc::GameState::validate(void) Validating GameState
W void __cdecl OpenApoc::GameState::validate(void) Validated GameState
W void __cdecl OpenApoc::Scenery::collapse(class OpenApoc::GameState &) Scenery at {100,76,3} type CITYTILE_CITYMAP_87 now falling
W void __cdecl OpenApoc::Scenery::collapse(class OpenApoc::GameState &) Scenery at {101,76,4} type CITYTILE_CITYMAP_88 now falling
W void __cdecl OpenApoc::Scenery::collapse(class OpenApoc::GameState &) Scenery at {102,76,3} type CITYTILE_CITYMAP_87 now falling
W void __cdecl OpenApoc::Scenery::collapse(class OpenApoc::GameState &) Scenery at {98,76,3} type CITYTILE_CITYMAP_87 now falling
W void __cdecl OpenApoc::Scenery::collapse(class OpenApoc::GameState &) Scenery at {99,76,4} type CITYTILE_CITYMAP_88 now falling
W void __cdecl OpenApoc::Scenery::collapse(class OpenApoc::GameState &) Scenery at {103,76,3} type CITYTILE_CITYMAP_73 now falling
E void __cdecl OpenApoc::VehicleMission::start(class OpenApoc::GameState &,class OpenApoc::Vehicle &) Restarting self destruct?

**To Reproduce**
load save, wait a few seconds

**Comment by JonnyH** [comment]
Thanks for the report!

There seem to be 2 issues I noticed running this save:

There's the printed error when VEHICLE_2076 (the alien transporter) hits the ground - I suspect it's related to the "sliding" mechanic trying to find a open spot but not sure at this point.

But then skipping past this error, when "Valkyrie Interceptor 1" gets hit a few seconds later it is destroyed, then there's an issue with how it kills the agents that are aboard:

Vehicle.cpp:1880

```
	// Dying will remove agent from current agents list
	for (auto agent : currentAgents)
	{
		agent->die(state, true);
	}
```

is invalid as Agent::die erases it from the Vehicle::currentAgents container, invalidating the iterator.

**Comment by kgd192** [comment]
> But then skipping past this error, when "Valkyrie Interceptor 1" gets hit a few seconds later it is destroyed, then there's an issue with how it kills the agents that are aboard:
> 
> Vehicle.cpp:1880
> 
> ```
> 	// Dying will remove agent from current agents list
> 	for (auto agent : currentAgents)
> 	{
> 		agent->die(state, true);
> 	}
> ```
> 
> is invalid as Agent::die erases it from the Vehicle::currentAgents container, invalidating the iterator.

This issue has been addressed in commit 1d35e9c will now beginn to investigate the actual issue with the alien ship

**Comment by FilmBoy84** [comment]
Thanks for confirming fixed 🍻 

Closing

**Comment by FilmBoy84** [comment]
Although, i cannot find the PR for this commit?
Can you submit for review when ready please?

**Comment by kgd192** [comment]
@FilmBoy84 Yeah i didnt put in the PR yet due to the outstanding fixes at least the Agent ID. Do you reckon we should split the rest into a different issue?

**Comment by FilmBoy84** [comment]
Could split them, yes

Would make review quicker and get the fix for the above merged promptly

But no worries if you would prefer to do as one

**Comment by andremmfaria** [comment]
Looking into fixing this issue I found issues #1613 and #1573. Those seem to have the same root cause of this issue. 
I have created #1648 to fix all 3. Will test once build finishes and I have some time to do so.
